lib/tools.js

```diff
@@ -1932,6 +1932,63 @@ function removeIdFromAllEnums(objects, id) {
     });
 }
 
+/**
+ * Validates types of obj.common properties and object.type, if invalid types are used, an error is thrown.
+ * If attributes of obj.common are not provided, no error is thrown. obj.type has to be there and has to be valid.
+ *
+ * @param {object} obj an object which will be validated
+ * @throws Error if a property has the wrong type or obj.type is non existing
+ */
+function validateGeneralObjectProperties(obj) {
+    if (!obj || obj.type === undefined) {
+        throw new Error(`obj.type has to exist`);
+    }
+
+    if (typeof obj.type !== 'string') {
+        throw new Error(`obj.type has an invalid type! Expected "string", received  "${typeof obj.type}"`);
+    }
+
+    const allowedObjectTypes = ['state', 'channel', 'device', 'enum', 'host', 'adapter', 'instance', 'meta', 'config', 'script', 'user', 'group', 'chart'];
+    if (!allowedObjectTypes.includes(obj.type)) {
+        throw new Error(`obj.type has an invalid value (${obj.type}) but has to be one of ${allowedObjectTypes.join(', ')}`);
+    }
+
+    // obj.common is optional in general check
+    if (!obj.common) return;
+
+    if (obj.common.name !== undefined && typeof obj.common.name !== 'string') {
+        throw new Error(`obj.common.name has an invalid type! Expected "string", received  "${typeof obj.common.name}"`);
+    }
+
+    if (obj.common.type !== undefined) {
+        if (typeof obj.common.type !== 'string') {
+            throw new Error(`obj.common.type has an invalid type! Expected "string", received  "${typeof obj.common.type}"`);
+        }
+
+        // if object type indicates a state, check that common.type matches
+        const allowedStateTypes = ['number', 'string', 'boolean', 'array', 'object', 'mixed', 'file'];
```

Comment:
I think "json" could be too.. Must check the "admin"

Reply:
Ok. no "json" type found.

Reply:
But I also remember from old SCHEMA.md that "jso"N was there as state type

Reply:
Hm there was also no `json` when I replaced SCHEMA.md today https://github.com/ioBroker/ioBroker/commit/ff4ad136ea06019309de8b336fff12fb324f59ed#diff-67a5df360d29ac792619774dd70c287eL200

Reply:
@Apollon77 but what would be the difference between json and object?

Reply:
good question :-) I think noone uses it ... so lets leave it out, and wen will see warnings